src/main/java/io/opencensus/integration/jdbc/OcWrapCallableStatement.java

```diff
@@ -14,548 +14,554 @@
 
 package io.opencensus.integration.jdbc;
 
+import io.opencensus.common.Scope;
 import io.opencensus.integration.jdbc.Observability.TraceOption;
+import io.opencensus.integration.jdbc.Observability.TrackingOperation;
+import java.sql.CallableStatement;
 import java.sql.SQLException;
 import java.util.EnumSet;
 
-public class OcWrapCallableStatement implements java.sql.CallableStatement {
-  private final java.sql.CallableStatement callableStatement;
+/**
+ * Wraps and instruments a {@link CallableStatement} instance with tracing and metrics using
+ * OpenCensus.
+ */
+public class OcWrapCallableStatement implements CallableStatement {
+  private final CallableStatement callableStatement;
   private final boolean shouldAnnotateSpansWithSQL;
 
-  public OcWrapCallableStatement(
-      java.sql.CallableStatement callableStatement, EnumSet<TraceOption> opts) throws SQLException {
+  public OcWrapCallableStatement(CallableStatement callableStatement, EnumSet<TraceOption> opts) {
     this.callableStatement = callableStatement;
     this.shouldAnnotateSpansWithSQL = Observability.shouldAnnotateSpansWithSQL(opts);
   }
 
   @Override
   public void addBatch() throws SQLException {
-    Observability.RoundtripTrackingSpan span =
+    TrackingOperation trackingOperation =
         Observability.createRoundtripTrackingSpan(
             "java.sql.CallableStatement.addBatch", "addBatch");
 
-    try {
+    try (Scope ws = trackingOperation.withSpan()) {
       this.callableStatement.addBatch();
     } catch (Exception e) {
-      span.recordException(e);
+      trackingOperation.endWithException(e);
       throw e;
     } finally {
-      span.close();
+      trackingOperation.end();
     }
   }
 
   @Override
   public void addBatch(String SQL) throws SQLException {
-    Observability.RoundtripTrackingSpan span =
+    TrackingOperation trackingOperation =
         Observability.createRoundtripTrackingSpan(
             "java.sql.CallableStatement.addBatch",
             "addBatch",
             this.shouldAnnotateSpansWithSQL,
             SQL);
 
-    try {
+    try (Scope ws = trackingOperation.withSpan()) {
```

Comment:
I think this is unnecessary since we are invoking trackingOperation.end in the finally clause.

Reply:
Not true. There are 2 different things here:
1) The span which is closed when call end().
2) This scope which deals with only installing the Span in the current context.